go_test.go

```diff
@@ -0,0 +1,171 @@
+				&Change{Name: "func xxxx", InsLines: 3, DelLines: 0, Inserted: true, Deleted: false, IsOther: false},
+				&Change{Name: "other", InsLines: 2, DelLines: 1, Inserted: false, Deleted: false, IsOther: true},
+			},
+			error(nil),
+		},
+		{ // 5
+			[]byte(fileA),
+			[]byte(fileBNewLines),
+			[]*Change{
+				&Change{Name: "func aaaa", InsLines: 3, DelLines: 1, Inserted: false, Deleted: false, IsOther: false},
+				&Change{Name: "func bbbb", InsLines: 3, DelLines: 1, Inserted: false, Deleted: false, IsOther: false},
+				&Change{Name: "func main", InsLines: 3, DelLines: 1, Inserted: false, Deleted: false, IsOther: false},
+				&Change{Name: "other", InsLines: 7, DelLines: 1, Inserted: false, Deleted: false, IsOther: true},
+			},
+			error(nil),
+		},
+		{ // 6
+			[]byte(fileBNewLines),
+			[]byte(fileA),
+			[]*Change{
+				&Change{Name: "func aaaa", InsLines: 1, DelLines: 3, Inserted: false, Deleted: false, IsOther: false},
+				&Change{Name: "func bbbb", InsLines: 1, DelLines: 3, Inserted: false, Deleted: false, IsOther: false},
+				&Change{Name: "func main", InsLines: 1, DelLines: 3, Inserted: false, Deleted: false, IsOther: false},
+				&Change{Name: "other", InsLines: 1, DelLines: 7, Inserted: false, Deleted: false, IsOther: true},
+			},
+			error(nil),
+		},
+		{ // 7
+			[]byte(fileBDelete),
+			[]byte(fileA),
+			[]*Change{
+				&Change{Name: "func aaaa", InsLines: 1, DelLines: 3, Inserted: false, Deleted: false, IsOther: false},
+				&Change{Name: "func bbbb", InsLines: 1, DelLines: 0, Inserted: true, Deleted: false, IsOther: false},
+				&Change{Name: "func main", InsLines: 1, DelLines: 3, Inserted: false, Deleted: false, IsOther: false},
+				&Change{Name: "other", InsLines: 1, DelLines: 1, Inserted: false, Deleted: false, IsOther: true},
+			},
+			error(nil),
+		},
+	}
+	for i, test := range tests {
+		t.Run(fmt.Sprintf("TEST %d/%d", i+1, len(tests)), func(t *testing.T) {
+			changes, err := Go(test.fileA, test.fileB)
+			if err != test.err {
+				t.Fatalf("expected %v, got %v", test.err, err)
+			}
+			lenc := len(changes)
+			lent := len(test.changes)
+			if lenc != lent {
+				t.Fatalf("expected %d changes, got %d changes", lent, lenc)
+			}
+			for i, c := range changes {
+				if !reflect.DeepEqual(*c, *test.changes[i]) {
+					t.Fatalf("expected %v, got %v", *test.changes[i], *c)
+				}
+			}
+		})
+	}
+}
+
+//------------------------------------------------------
+// test files
+
+const fileA = `package main
+//comment
+func main() {}
+func aaaa() {}
+func bbbb() {}`
+
+const fileB = `package main
+func main() {
+	return
+}
+//comment
+func aaaa() {
+	return
+}
+//comment
+func bbbb() {
+	return
+}`
+
+const fileBNewLines = `package main
+
+
+func main() {
+	return
+}
+
+//comment
+
+func aaaa() {
+	return
+}
+
+//comment
+func bbbb() {
+	return
+}`
+
+const fileBRename = `package main
+func main() {
+	return
+}
+//comment
```

Comment:
Using identical comments here makes it possible to have bugs slip through the cracks. Change to `// xxxx docs`, `// bbbb docs`, etc? And add comments at end as well. I see that you're using blank lines to trigger "other" additions and deletions, but I think it'd be easier to spot that in the tests if you changed to:

```
// xxxx docs
// xxxx added line
```

or some such.